samtranslator/internal/resource_validator.py

```diff
@@ -1,18 +1,32 @@
 """A resource validator to help validate resource properties and raise exception when some value is unexpected."""
-from typing import Any, Dict, Type, TypeVar
+from typing import Any, Dict, List, Type, TypeVar
 
 from pydantic import BaseModel
+from pydantic.error_wrappers import ValidationError
+
+from samtranslator.model.exceptions import InvalidResourceException
 
 T = TypeVar("T", bound=BaseModel)
 
 
+def _parse_exception_msg(errors: List[Any]) -> str:
+    try:
+        error_properties = ", ".join([error["loc"][0] for error in errors])
+        return f"Given resource property '{error_properties}' is invalid"
+    except KeyError:
+        return "Given resource property is invalid"
+
+
 # Note: For compabitliy issue, we should ONLY use this with new abstraction/resources.
-def to_model(resource_properties: Dict[Any, Any], cls: Type[T]) -> T:
+def validate_and_return_model(logical_id: str, resource_properties: Dict[Any, Any], cls: Type[T]) -> T:
```

Comment:
At this point why not have it part of a `Resource`? Since it's so coupled to `logical_id`.